Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/Annotation/AddIshAnnotation.cs

```diff
@@ -0,0 +1,345 @@
+    /// <example>
+    /// <code>
+    /// $ishSession = New-IshSession -WsBaseUrl "https://example.com/InfoShareWS/" -IshUserName "username" -IshUserPassword  "userpassword"
+    /// $ishAnnotation = Add-IshAnnotation -IshSession $ishsession `
+    ///                     -PubLogicalId "GUID-MYPUBLICATIONLOGICALID" `
+    ///                     -PubVersion "1" `
+    ///                     -LogicalId "MYCONTENTOBJECTLOGICALID" `
+    ///                     -Version "1" `
+    ///                     -Lng "en" `
+    ///                     -Type "General" `
+    ///                     -Text "My annotation text" `
+    ///                     -Status "VANNOTATIONSTATUSUNSHARED" `
+    ///                     -Category "Comment" `
+    ///                     -Address "My annotation address" `
+    ///	$ishAnnotation = $ishAnnotation | Set-IshMetadataField -Name "FISHANNOTATIONTEXT" -Level Annotation -Value "My annotation text updated"
+    /// $ishAnnotationAdded = $ishAnnotation | Add-IshAnnotation -IshSession $ishsession
+    /// </code>
+    /// <para>Add annotation providing IshAnnotation object.</para>
+    /// </example>
+    [Cmdlet(VerbsCommon.Add, "IshAnnotation", SupportsShouldProcess = false)]
+    [OutputType(typeof(IshAnnotation))]
+    public sealed class AddIshAnnotation :AnnotationCmdlet
+    {
+        /// <summary>
+        /// <para type="description">The IshSession variable holds the authentication and contract information. This object can be initialized using the New-IshSession cmdlet.</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParametersGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshAnnotationGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "MetadataGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshSession IshSession { get; set; }
+
+        /// <summary>
+        /// <para type="description">Publication LogicalId</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParametersGroup")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectGroup")]
+        [ValidateNotNullOrEmpty]
+        public string PubLogicalId { get; set; }
+
+        /// <summary>
+        /// <para type="description">Publication version</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParametersGroup")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectGroup")]
+        [ValidateNotNullOrEmpty]
+        public string PubVersion { get; set; }
+
```

Comment:
An extra field FISHPUBLANGUAGE is added for SP2 which must be filled in other the Annotation will not be visible. So add an extra mandatory parameter PubLng similar to PubVersion. If the parameter is set for a previous version (< 14.0.2) then we will log a warning indicating that this parameter will be ignored.